lib/ansible/modules/system/user.py

```diff
@@ -2638,7 +2642,7 @@ def main():
     if err:
         result['stderr'] = err
 
-    if user.user_exists():
+    if user.user_exists() and user.state == 'present':
```

Comment:
We should only fetch use information with state `present`, there is no need to fetch use info if we are deleting a user.

Reply:
If `user.state == 'absent'` the user account will be removed, which means `user.user_exists()` will be `False`. No need for the additional condition.

Reply:
But it doesn't hurt to have this check, because we had an issue because of it. We had local users on our hosts and than we added those hosts into a FreeIPA domain. The FreeIPA domain had the same exact users that were created locally before, so we deleted our local users using this module, so what happened was that this condition returned that the user did exist (because it found a user on the FreeIPA domain) and than it created that user's home dir.

For example if there is a `kustodian` local user and the same `kustodian` user exists on the FreeIPA domain (or any central user management system), when you run:

```
- user:
    name: kustodian
    state: absent
```
This would delete the local `kustodian` user and than it would create the remote `kustodian` user home dir, which in our case was `/home/ipa.lan/kustodian`. Which is not what you expect from the `user` module to do when it deletes a user. Also it created the home dirs with `755`, which we also don't want (`useradd` creates them with 700). I know this is an edge case, but its something that can happen (it happened to us) and this additional condition solves the problem.

Reply:
I see. Thank you for that great explanation. Since it doesn’t hurt to have it plus fixes the scenario you describe, we should keep it.